src/index.js

```diff
@@ -7,40 +7,31 @@ export default function (source, inputSourceMap) {
 
   const webpackRemainingChain = loaderUtils.getRemainingRequest(this).split('!')
   const filename = webpackRemainingChain[webpackRemainingChain.length - 1]
-  
+
   const rollupConfig = {
     entry: {
       path: filename,
       contents: source
     },
-    plugins: [
-      memory()
-    ],
     external(id) {
       return !(/\.(js|es6)$/.test(id))
     }
   }
 
-  const rollupOptions = this.options.rollup
-  if (rollupOptions) {    
-    if (Array.isArray(rollupOptions)) {
-      rollupConfig.plugins = rollupConfig.plugins.concat(rollupOptions)
-    } else {
-      rollupConfig.plugins = rollupConfig.plugins.concat(rollupOptions.plugins)
-      delete rollupOptions.plugins
-      Object.assign(rollupConfig, rollupOptions)
-    }
-  }
+  const parsed = loaderUtils.parseQuery(this.query)
+  const plugins = this.options.rollup
+  const options = plugins ? { plugins } : parsed
+
+  Object.assign(rollupConfig, options)
+  rollupConfig.plugins = [memory(), ...options.plugins || []]
 
```

Comment:
\cc @TheLarkInn. Is this bananas or an OK way to go about supporting these [two config flavors](https://github.com/egoist/rollup-loader#usage).